Werewolf for Telegram/Werewolf Control/Helpers/Bot.cs

```diff
@@ -509,10 +510,24 @@ public static Node GetBestAvailableNode()
         }
 
 
-        internal static Task<Message> Send(string message, long id, bool clearKeyboard = false, InlineKeyboardMarkup customMenu = null, ParseMode parseMode = ParseMode.Html, int? messageThreadId = null)
+        internal static Task<Message> Send(string message, long id, bool clearKeyboard = false, InlineKeyboardMarkup customMenu = null, ParseMode parseMode = ParseMode.Html, int? messageThreadId = null, bool forceTopic = true)
         {
             //MessagesSent++;
             //message = message.Replace("`",@"\`");
+
+            // Try to load GroupTopicId from the database if no thread ID is provided
+            if (messageThreadId == null && forceTopic)
+            {
+                using (var db = new WWContext())
+                {
+                    var group = db.Groups.FirstOrDefault(g => g.GroupId == id);
+                    if (group?.GroupTopicId != null)
+                    {
+                        messageThreadId = group.GroupTopicId;
+                    }
+                }
+            }
+
```

Comment:
I fail to see the benefit that the `forceTopic` parameter brings...? Why can't we just call `Send` with the appropriate `messageThreadId` value in all places? Here we're again opening a database connection whenever `Send` is called.

If the current command's handler just passes the topic ID of the received message to `Send`, we're good (because messages coming from a different topic won't go through to the command handler anyway). No need to check the database for the group topic ID again.

Or am I missing something here?

Reply:
I was using forceTopic on some places to allow bot to reply message in general topic which have `null` in thread id field. 

It is only helpful to response to admin. Will remove it as we are going to trust given thread id